iib/workers/tasks/build_regenerate_bundle.py

```diff
@@ -529,3 +540,31 @@ def _apply_repo_enclosure(bundle_metadata, org_enclose_repo_namespace, org_enclo
         replacement_pullspecs[pullspec] = new_pullspec
 
     _replace_csv_pullspecs(bundle_metadata, replacement_pullspecs)
+
+
+def _get_related_bundle_images(bundle_metadata, request_id):
+    """
+    Get bundle images in the CSV files of the bundle being regenerated and store them in a file.
+
+    :param dict bundle_metadata: the dictionary of CSV's and relatedImages pull specifications
+    :param int request_id: the ID of the IIB build request
+    """
+    worker_config = get_worker_config()
+    related_bundles_dir = worker_config['iib_request_related_bundles_dir']
+    related_bundles_file = os.path.join(related_bundles_dir, f'{request_id}_related_bundles.json')
+
+    if os.path.isfile(related_bundles_file):
+        log.debug('Related bundles already exist for this request. Skipping...')
```

Comment:
Why not overwrite it? It seems like if we're calling this method again, something has changed and we want to regenerate the list of related images.

Reply:
Because it won't always work. As you mentioned above, the managed services use case won't have the images present in the namespace yet.

Reply:
Given the new approach, should this overwrite now?